CMakeLists.txt

```diff
@@ -465,6 +465,22 @@ elseif(HPX_WITH_PARALLEL_LINK_JOBS)
   hpx_warn("Job pooling is only available with Ninja generators.")
 endif()
 
+hpx_option(
+  HPX_HAVE_SIMD_BACKEND
+  STRING
+  "Simd Backend to use for vectorization. Defaults to OMP, other possible options are NONE (compiler assisted vectorization)"
+  "OMP"
+  CATEGORY "Build Targets"
+)
+
+if(NOT HPX_HAVE_SIMD_BACKEND)
+  set(HPX_HAVE_SIMD_BACKEND "OMP")
```

Comment:
This will force using the `-fopenmp-simd` even on platforms that are not based on gcc or clang. See also https://github.com/STEllAR-GROUP/hpx/actions/runs/5597037732/job/15160508847?pr=6303 for some evidence.

Reply:
What other options except `"OMP"` would the user have?